website/docs/history-and-hard-forks/history-vm-version.mdx

```diff
@@ -0,0 +1,212 @@
+
+1. Cycles for RISC-V instructions are determined based on real hardware that implement RISC-V ISA.
+2. Cycles for syscalls are measured based on real runtime performance metrics obtained while benchmarking current CKB implementation.
+
+**Initial Loading Cycles**
+
+For each byte loaded into CKB VM in the initial ELF loading phase, 0.25 cycles will be charged. This is to encourage dapp developers to ship smaller smart contracts as well as preventing DDoS attacks using large binaries. Notice fractions will be rounded up here, so 30.25 cycles will become 31 cycles.
+
+**instructions Cycles**
+
+All instructions consume 1 cycle except the following ones:
+
+| Instruction | Cycles               |
+| ----------- | -------------------- |
+| JALR        | 3                    |
+| JAL         | 3                    |
+| J           | 3                    |
+| JR          | 3                    |
+| BEQ         | 3                    |
+| BNE         | 3                    |
+| BLT         | 3                    |
+| BGE         | 3                    |
+| BLTU        | 3                    |
+| BGEU        | 3                    |
+| BEQZ        | 3                    |
+| BNEZ        | 3                    |
+| LD          | 2                    |
+| SD          | 2                    |
+| LDSP        | 2                    |
+| SDSP        | 2                    |
+| LW          | 3                    |
+| LH          | 3                    |
+| LB          | 3                    |
+| LWU         | 3                    |
+| LHU         | 3                    |
+| LBU         | 3                    |
+| SW          | 3                    |
+| SH          | 3                    |
+| SB          | 3                    |
+| LWSP        | 3                    |
+| SWSP        | 3                    |
+| MUL         | 5                    |
+| MULW        | 5                    |
+| MULH        | 5                    |
+| MULHU       | 5                    |
+| MULHSU      | 5                    |
+| DIV         | 32                   |
+| DIVW        | 32                   |
+| DIVU        | 32                   |
+| DIVUW       | 32                   |
+| REM         | 32                   |
+| REMW        | 32                   |
+| REMU        | 32                   |
+| REMUW       | 32                   |
+| ECALL       | 500 (see note below) |
+| EBREAK      | 500 (see note below) |
+
+**Syscall Cycles**
+
+As shown in the above chart, each syscall will have 500 initial cycle consumptions. This is based on real performance metrics gathered benchmarking CKB implementation, certain bookkeeping logics are required for each syscall here.
+
+In addition, for each byte loaded into CKB VM in the syscalls, 0.25 cycles will be charged. Notice fractions will also be rounded up here, so 30.25 cycles will become 31 cycles.
+
+## VM 1
+
+introduced from **1st Hard Fork – CKB Edition Mirana (2021)**
```

Comment:
```suggestion
Introduced since [1st Hard Fork – CKB Edition Mirana (2021)](/docs/history-and-hard-forks/ckb-hard-fork-history#1st-hard-fork--ckb-edition-mirana-2021)
```